### PR TITLE
feat: Add Login,Logout and Signup

### PR DESCRIPTION
**Pull request by ravikumar0403**
### change log
---
- Add login page
- Add signup page
- create protected routes
- add logout in navbar

### CODE REVIEW

File: src/config/routes.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import { History, Layout, LikedVideos, Playlist, WatchLater } from "components";
-import { ExplorePage, Homepage } from "pages";
+import { RequireAuth } from "components/RequireAuth";
+import MockmanEs from "mockman-js";
+import { ExplorePage, Homepage, Login, Signup } from "pages";
 
 export const routes = [
```
Comment (sonishreyas) [inline src/config/routes.js:6]: Really liked how you have defined the routes

File: src/context/StateProvider.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+import { AuthProvider } from "./auth-context";
 import { CategoriesProvider } from "./categories-context";
 import { VideosProvider } from "./videos-context";
 
 export const StateProvider = ({ children }) => {
   return (
-    <CategoriesProvider>
-      <VideosProvider>{children}</VideosProvider>
-    </CategoriesProvider>
+    <AuthProvider>
```
Comment (sonishreyas) [inline src/context/StateProvider.js:7]: Nice way to get all the providers

**Review by sonishreyas — APPROVED** [review]
The code looks clean and easy to understand.

File: src/hooks/useDocumentTitle.js [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { useEffect } from "react";
+
+export const useDocumentTitle = (title) => {
```
Comment (yashpatel1797) [inline src/hooks/useDocumentTitle.js:3]: nice use of a hook to change the title.

**Review by yashpatel1797 — APPROVED** [review]
Nice folder structure.
code is clean and easy to understand.
like how you have created routes as well.

**Pull request merged**